nixos/modules/services/web-servers/traefik.nix

```diff
@@ -0,0 +1,78 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.traefik;
+  configFile = pkgs.writeText "traefik.toml" cfg.config;
+in {
+  options.services.traefik = {
+    enable = mkEnableOption "Traefik web server";
+
+    config = mkOption {
+      default = "";
+      example = ''
```

Comment:
I agree with @globin that the using remarshal to convert nix to toml would be an a nice enhancement here.